### DISCUSSION TITLE
How do I use the keepSourceTokens option in the parser

### DISCUSSION THREAD

**Discussion by nfstein** [eemeli/yaml#404] [Q&A] [answered]
When parsing a yaml file I am trying to use the `keepSourceTokens` option to get tokens in my resulting object. I dont see any difference in the object produced with or without the `keepSourceTokens` option. Based on the documentation [here](https://eemeli.org/yaml/#options)  I was expecting a `srcToken` key in the resulting json. My end goal is to have position information in the resulting json for the various keys and values. Am I missing something here or is there a better way to get the position information.
```
import fs from 'fs';
import { LineCounter, parse } from 'yaml';


function parseYamlTest(filePath) {
  const lineCounter1 = new LineCounter(); 
  const lineCounter2 = new LineCounter(); 

  const file = fs.readFileSync(filePath);
  let parsedWithSource = parse(file.toString(), { lineCounter1, keepSourceTokens: true} );
  console.log(parsedWithSource)
  let parsedNoSource = parse(file.toString(), { lineCounter2} );
  console.log(parsedNoSource)

}

parseYamlTest("./test.yaml");
```

**Comment by eemeli** [comment, marked as answer]
The information you're looking for is included in the output of `parseDocument()` rather than `parse()`.
